core/runtime/jsruntime.rs

```diff
@@ -1533,20 +1521,23 @@ impl JsRuntime {
         let mut has_evaluated = false;
 
         // Fast main realm poll
-        let main_realm = self.inner.main_realm.as_ref().unwrap();
-        loop {
-          let poll_imports =
-            main_realm.prepare_dyn_imports(&mut self.inner.v8_isolate, cx)?;
-          assert!(poll_imports.is_ready());
-
-          let poll_imports =
-            main_realm.poll_dyn_imports(&mut self.inner.v8_isolate, cx)?;
-          assert!(poll_imports.is_ready());
-
-          if main_realm.evaluate_dyn_imports(&mut self.inner.v8_isolate) {
-            has_evaluated = true;
-          } else {
-            break;
+        {
+          let main_realm = self.inner.main_realm.as_ref().unwrap();
+          let mut scope = main_realm.handle_scope(&mut self.inner.v8_isolate);
+          let modules = main_realm.0.module_map();
```

Comment:
We will clean this up in a future PR. Right now we just moved methods and tweaked the scope a bit.